### PR TITLE
ENH Use the new algolia search index and API

### PR DESCRIPTION
**Pull request by GuySartorelli**
## What's in this PR?
1. Update dev environment variables (note: these get exposed in the browser so they're not secret - it's safe to add them here)
2. Add note about how to update algolia crawler when adding a new major release
3. Replace the old v2 docsearch API usage with the new v3 docsearch API usage, using the new shiny react component
  - Note: The old API took an input element ID, which meant we could style the thing however we wanted. The new API takes a container and completely replaces the container _even if you're not using the react component_, so our styling options are limited. There is [a page in algolia docs for styling](https://docsearch.algolia.com/docs/styling) but it has no useful information in it
4. Update styling for the new search component so it behaves nicely with our header and breakpoints

## Action required after merging
After this is merged, we need to update the environment variables in netlify for the main docs. I'm happy to do that - but make sure I'm ready to do so when you merge this so we don't end up with broken search for an extended period.

## Issue
- https://github.com/silverstripe/doc.silverstripe.org/issues/259

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -23,14 +23,9 @@ const Header: StatelessComponent<HeaderProps> = ({ handleSidebarToggle }): React
...
-        const otherNode = getNodes().find(n => n.slug === newPath);
-        // This has to be a hard refresh, because the sidebar needs to unmount
-        if (otherNode) {
-          navigate(otherNode.slug);
-        } else {
-          navigate(`/en/${ver}`);
-        }
+        const newPath = getVersionPath(currentNode, ver);
+        // This has to be a hard refresh, because the sidebar and searchbar need to unmount
+        navigate(newPath);
```
Comment (GuySartorelli) [inline src/components/Header.tsx:28]: This ultimately still does the same thing it was doing before - it's just a lot cleaner.

File: src/components/Layout.tsx [hunk 2]
```diff
@@ -20,18 +20,21 @@ const Layout: StatelessComponent<LayoutProps> = ({ children, pageContext: { slug
   const currentNode = getCurrentNode();
   const versionMessage = getVersionMessage();
 
+  const helmetLinks = [{
+    rel: 'preconnect',
+    href: `https://${process.env.GATSBY_DOCSEARCH_APP_ID}-dsn.algolia.net`,
+    crossOrigin: ''
```
Comment (GuySartorelli) [inline src/components/Layout.tsx:26]: See https://docsearch.algolia.com/docs/DocSearch-v3#performance-optimization

File: src/components/SearchBox.tsx [hunk 3]
```diff
@@ -1,75 +1,51 @@
 import React from 'react'
-import { StatelessComponent, ReactElement, useEffect, useState } from 'react';
-import { navigateTo } from "gatsby-link"
+import { StatelessComponent, ReactElement } from 'react';
+import { navigate } from "gatsby"
```
Comment (GuySartorelli) [inline src/components/SearchBox.tsx:3]: `navigateTo` and `navigate` do the same thing, but `navigateTo` is deprecated so may as well stop using that while we're at it.

File: src/components/SearchBox.tsx [hunk 4]
```diff
@@ -1,75 +1,51 @@
...
+          navigate: ({ itemUrl }) => navigate(makeUrlRelative(itemUrl)),
+        }}
+        // Overrides the link container for search results so we can use gatsby navigation instead of full page loads
+        hitComponent={({ hit, children }) => {
+          const relativeUrl = makeUrlRelative(hit.url);
+          // We can't use a gatsby <Link> component here - for some reason it makes the links disappear on hover
+          return <a onClick={(e) => handleClick(e, relativeUrl)} href={relativeUrl}>{children}</a>
+        }}
+      />
+    );
```
Comment (GuySartorelli) [inline src/components/SearchBox.tsx:48]: See https://docsearch.algolia.com/docs/api

**Review by emteknetnz — APPROVED** [review]
Code looks fine, self-merge when you're ready

**Pull request merged**